### PR TITLE
feat: add Vagrantfile for local development

### LINKED ISSUE #66: Finalize Development Environment
Gold image with required tools. Specify install of dependencies in vagrantfile. 

This prevents drift across environments while developing.

Ubuntu LTS.

### PR DESCRIPTION
**Pull request by jasonwalsh**
### Purpose

The objective of [Vagrant](https://www.vagrantup.com/) is to replicate production environments. Our production environment uses AWS Lambda, which executes source code in Docker runtimes.

Vagrant ensures that our development environment is consistent and helps prevent potential drift like dependency mismatches.

### Dependencies

- [Vagrant](https://www.vagrantup.com/downloads.html)
- [VirtualBox](https://www.virtualbox.org/wiki/Downloads)

### Usage

    $ vagrant up

If the `vagrant up` process is successful, then invoke the following command:

    $ vagrant ssh

The `/vagrant` directory is synced to the root of the [hopdrops/hopdrops](https://github.com/hopdrops/hopdrops) repository.

### Building

To invoke the AWS SAM [build](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-build.html) command, invoke the following command:

    $ sam build --use-container

To start the local API, invoke the following command:

    $ sam local start-api --host 0.0.0.0

**Note:** The `--host` flag is important, otherwise `sam` listens on `localhost` and is not accessible via a web browser on the host machine (e.g., the machine that invoked `vagrant up`).

Closes #66

### CODE REVIEW

**Comment by jasonwalsh** [comment]
It is also worth mentioning that the `box_url` in the `Vagrantfile` comes from the hopdrops S3 bucket. The [hopdrops/vagrant](https://github.com/hopdrops/vagrant) repository contains the provisioning scripts for configuring the development environment.

Right now the process of packaging the Vagrant box and uploading it to S3 is *not* automated, although it is worth consideration in the future to help streamline development.

File: template.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ Description: >
 
 Globals:
   Function:
-    Runtime: python3.7
+    Runtime: python3.6
```
Comment (jasonwalsh) [inline template.yaml]: The Vagrant box has `python3.6` installed and was causing the `sam build` command to fail with the following error:

```Error: PythonPipBuilder:None - Binary validation failed!```

See aws/aws-toolkit-jetbrains#828 for more details.

Comment (AmillerSC) [reply template.yaml]: I can confirm this works fine for me now

**Comment by brh55** [comment]
Add these commands to README as well

**Pull request merged**